pkg/apis/storage/validation/validation.go

```diff
@@ -79,6 +81,12 @@ func validateParameters(params map[string]string, fldPath *field.Path) field.Err
 		if len(k) < 1 {
 			allErrs = append(allErrs, field.Invalid(fldPath, k, "field can not be empty."))
 		}
+		if k == "zone" {
```

Comment:
You can't check all StorageClass instances here, there may be an external or flex provisioner that interprets "zone" parameter in a very different way. All checks must be done in volume plugins (and they're already there, I think).